openmeetings-server/src/site/markdown/LdapAndADS.md

```diff
@@ -0,0 +1,238 @@
+
+# LDAP and Active Directory Integration
+
+This manual explains the configuration to use an already existing
+(e.g. corporate or university) LDAP/AD Server for user authentication
+into OpenMeetings. Successful integration has also been reported with
+Novell eDirectory.
+
+For installation and setup instructions of an LDAP server itself see
+https://help.ubuntu.com/lts/serverguide/openldap-server.html
+
+## LDAP Configuration in OpenMeetings
+
+To configure access to an LDAP Server, three things are needed:
+
+### 1. Information how to access the LDAP Server for user lookup
+
+-   Obtain the ip address and port number of the LDAP server. For the
+    example configuration in the next section we will use the ip
+    address 10.8.0.10 and the default LDAP port number 389.
+
+-   A single LDAP user account is needed that has read privileges to the
+    user entries that are to be used for login by OpenMeetings. Often
+    these accounts have the organizational unit attribute
+    `ou=system-users`.
+
+-   Obtain the cn and password of that user from the LDAP
+    administrator. For the example configuration we will use
+    `cn=openmeetings` and `password=12345`.
+
+-   Obtain the dc values of the root of the LDAP database. In our
+    example we will use `dc=mydomain,dc=org`.
+
+### 2. An LDAP config file
+
+-   This file needs to be located in the folder
+    `$OM_HOME/webapps/openmeetings/data/conf/` <sup><a id="fnr.1" class="footref" href="#fn.1">1</a></sup>.
+    An example file called `om_ldap.cfg` should come with your
+    OpenMeetings installation.
+
+-   The file needs to contain the access information obtained in the
+    previous section. For our example the file would contain the
+    following:
+
+    ```
+    ldap_conn_host=10.8.0.10
+    ldap_conn_port=389
+    ldap_conn_secure=false
+    
+    # The following lines specify the data of the ldap user with 
+    # read privileges to the user entries according to the LDAP DB.
+    ldap_admin_dn=cn=openmeetings,ou=system-users,dc=mydomain,dc=org
+    ldap_passwd=12345
+    
+    # The following line specifies where to search for users wanting to login 
+    ldap_search_base=ou=users,dc=mydomain,dc=org
+    
+    # The attribute in the LDAP DB which will be matched against the 
+    # given login data ("Username or mail address" field) in OpenMeetings:
+    ldap_search_query=(uid=%s)
+    
+    # the scope of the search might be: OBJECT, ONELEVEL, SUBTREE
+    ldap_search_scope=ONELEVEL
+    
+    # Ldap auth type(NONE, SEARCHANDBIND, SIMPLEBIND)
+    #  When using SIMPLEBIND a simple bind is performed on the LDAP server to check user authentication
+    #  When using NONE, the Ldap server is not used for authentication
+    ldap_auth_type=SIMPLEBIND
+    
+    # userDN format, will be used to bind if ldap_auth_type=SIMPLEBIND
+    # might be used to get provisionningDn in case ldap_auth_type=NONE
+    ldap_userdn_format=uid=%s,ou=users,dc=mydomain,dc=org
```

Comment:
With `SIMPLEBIND`,

LDAP `DN` will be constructed according to format provided
And authentication attempt will be made using password entered by user ...

Reply:
Isn't that what one would expect?

Reply:
Actually It depends :)))

there is `SEARCHANDBIND` mode in this case
1. user `DN` will be searched first (using admin credentials, search base and search query)
2. then this `DN` and user provided password will be used to authenticate

This might be useful in some cases

Sometimes it is just impossible to use `SIMPLEBIND` due to there is no common user format ....

Reply:
ok, thanks. Just to be sure (before my rewrite): The ldap_admin_dn and ldap_passwd is not necessary in case there is a common user format? Does the ldap server authenticate against a user and his password without the ldap_admin_dn and ldap_passwd provided?

Reply:
In case of `SIMPLEBIND`: yes